push.go

```diff
@@ -53,16 +53,53 @@ func (cmd *PushCommand) Run() error {
 type Sources []*Source
 
 type Source struct {
-	File          string      `yaml:"file,omitempty"`
-	ProjectID     string      `yaml:"project_id,omitempty"`
-	AccessToken   string      `yaml:"access_token,omitempty"`
-	FileFormat    string      `yaml:"file_format,omitempty"`
-	Params        *phraseapp.UploadParams `yaml:"params"`
+	File          string
+	ProjectID     string
+	AccessToken   string
+	FileFormat    string
+	Params        *phraseapp.UploadParams
 
 	RemoteLocales []*phraseapp.Locale
 	Extension     string
 }
 
+func (src *Source) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	m := map[string]interface{}{}
+	if err := unmarshal(m); err != nil {
+		return err
+	}
+
+	var ok bool
+	cfgErrStr := "configuration key %q has invalid value\nsee https://phraseapp.com/docs/developers/cli/configuration/"
+	for k, v := range m {
+		switch k {
+		case "file":
+			if src.File, ok = v.(string); !ok { return fmt.Errorf(cfgErrStr, k) }
```

Comment:
I would move all those string tests in a helper like `validateIsString` which returns the `cfgErrStr` thingy